scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -356,16 +356,29 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
           token.is[KwDo] || token.is[KwCatch] || token.is[KwFinally] || token.is[KwYield] ||
           token.is[KwMatch]
 
+        def outdent() = {
+          sepRegions = sepRegions.tail
+          curr = TokenRef(
+            new Indentation.Outdent(token.input, token.dialect, token.start, token.end),
+            curr.pos,
+            curr.pos,
+            prevPos
+          )
+          true
+        }
+
+        def existParenRegion: Boolean = sepRegions
+          .collectFirst {
+            case RegionParen => true
+            case other if !other.isIndented => false
+          }
+          .getOrElse(false)
+
         sepRegions match {
           case region :: tail if region.isIndented && canEndIndentation(curr.token) =>
-            sepRegions = sepRegions.tail
-            curr = TokenRef(
-              new Indentation.Outdent(token.input, token.dialect, token.start, token.end),
-              curr.pos,
-              curr.pos,
-              prevPos
-            )
-            true
+            outdent()
+          case _ if existParenRegion && token.is[Comma] =>
```

Comment:
:thinking: There is a similar handler in `LazyTokenIterator` https://github.com/scalameta/scalameta/blob/ea3c552a3a056a22586a7129d0d9c246c5fa672e/scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala#L419-L422

It looks like a better place for fix. The only problem I don't get why it doesn't work.

Reply:
`headOption` is the reason, there might be mutliple indented regions - I will fix it there